profile.py

```diff
@@ -80,6 +80,19 @@ def x_at_y(self, y, reverse=False):
         # B) if y < min(self.y) then condition self.y >= y will be satisfied on first item
         # np.argmax(cond) will be equal 0,
         # to exclude situation that y_handle[0] = y we also check if y < y_handle[0]
+
+        ## Consider this case: profiles values are descending from i.e. 5 to 1,
+        ## where f(0)=5, f(1)=4 ... f(4)=1
+        ## What would function x_at_y(2) return?
+
+        ## since ind==0 and 1 < 5 the if statement below is true and returned value
+        ## is nan. While the correct answer is 3 since f(3)=1.
+
+        ## Check main() function with the profile i entered there.
+        ## Also since the function returns x value given y value,
+        ## names of variable used in loop should be y not x (in my opinion)
+
+
```

Comment:
Read carefully docstring. 
We assume that the profile is first (more-less) rising then (more-less) falling down.
This function is doing lookup "from the left", checking if correspoing x is found on the (left) rising slope.
In the case of "descending profile" no such x is found and it is correct.
Try to call your function with reverse=True.

You are anyway right - better documentation is needed